include/aws/common/log.h

```diff
@@ -0,0 +1,120 @@
+#ifndef AWS_COMMON_LOG_H
+#define AWS_COMMON_LOG_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <stdarg.h>
+
+#include <aws/common/memory_pool.h>
+
+enum aws_log_level
+{
+    AWS_LOG_LEVEL_OFF = 0, /* No logs will be processed at all. */
+    AWS_LOG_LEVEL_FATAL,
+    AWS_LOG_LEVEL_ERROR,
+    AWS_LOG_LEVEL_WARN,
+    AWS_LOG_LEVEL_INFO,
+    AWS_LOG_LEVEL_DEBUG,
+    AWS_LOG_LEVEL_TRACE

```

Comment:
Do we want to let the user set the aws_log_level at runtime?

Users could provide it to the _init() function, and/or provide a function so they could toggle it at at any time.

Reply:
If we don't let the user adjust this at run-time there is the benefit that the entire AWS_LOG macro will get compiled away, since the internal if-check will have a constant expression of 0. Personally I like this idea. Would be nice to hear opinions from others though.

Reply:
We did this with the C++ SDK and I've yet to see a user take advantage of it. I'm fine optimizing for being able to compile it away.